ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java

```diff
@@ -36,25 +36,30 @@
 import org.hyperledger.besu.ethereum.mainnet.feemarket.BaseFeeMarket;
 import org.hyperledger.besu.ethereum.mainnet.feemarket.FeeMarket;
 
-import java.util.AbstractMap;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Comparator;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 import java.util.stream.LongStream;
 import java.util.stream.Stream;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
 import com.google.common.collect.Streams;
 
 public class EthFeeHistory implements JsonRpcMethod {
   private final ProtocolSchedule protocolSchedule;
   private final Blockchain blockchain;
+  private final Cache<RewardCacheKey, List<Wei>> cache;
+  private static final int MAXIMUM_CACHE_SIZE = 100_000;
+
+  record RewardCacheKey(Long blockNumber, List<Double> rewardPercentiles) {}
```

Comment:
can we do this by blockhash instead?  otherwise we might cache a non-canonical reward after a reorg.

Reply:
Good catch, let me review this part

Reply:
Done